tests/unit/test_routes.py

```diff
@@ -172,29 +208,101 @@ def test_get_api_key(module_client, module_db, fake_auth_from_oc):
     assert (isinstance(response.json['data'].get('apikey'), str))
 
 
+def test_get_api_key(module_client, module_db, fake_invalid_auth_from_oc):
+    client = module_client
+
+    response = client.post('api/v1/apikey',
+        follow_redirects=True,
+        json = dict(
+            email="test@example.org",
+            password="invalidpassword"
+    ))
+
+    assert (response.status_code == 401)
+
+
 def test_create_resource(module_client, module_db, fake_auth_from_oc):
     client = module_client
 
+    # Happy Path
     apikey = get_api_key(client)
     response = create_resource(client, apikey)
     assert (response.status_code == 200)
     assert (isinstance(response.json['data'].get('id'), int))
     assert (response.json['data'].get('name') == "Some Name")
 
+    # Invalid API Key Path
+    response = create_resource(client, "invalidapikey")
+    assert (response.status_code == 401)
+
 
 def test_update_resource(module_client, module_db, fake_auth_from_oc):
     client = module_client
 
+    # Happy Path
     apikey = get_api_key(client)
 
     response = client.put("/api/v1/resources/1",
-        json = dict(name="New name"),
+        json = dict(
+            name="New name",
+            languages=["New language"],
+            category="New Category",
+            url="https://new.url",
+            paid=False,
+            notes="New notes"
+        ),
         headers = {'x-apikey': apikey}
     )
 
     assert (response.status_code == 200)
     assert (response.json['data'].get('name') == "New name")
 
+    # Resource not found
+    response = client.put("/api/v1/resources/0",
+        json = dict(name="New name"),
+        headers = {'x-apikey': apikey},
+        follow_redirects = True
+    )
+    assert (response.status_code == 404)
+
+
+def test_commit_errors(module_client, module_db, fake_auth_from_oc, fake_commit_error):
+    client = module_client
+    apikey = get_api_key(client)
+
+    response = client.put("/api/v1/resources/1",
+        json = dict(name="New name"),
+        headers = {'x-apikey': apikey}
+    )
+    assert (response.status_code == 500)
+
+    response = create_resource(client, apikey)
+    assert (response.status_code == 500)
+
+
+def test_key_query_error(module_client, module_db, fake_auth_from_oc, fake_key_query_error):
+    client = module_client
+    response = client.post('api/v1/apikey', json = dict(
+        email="test@example.org",
+        password="supersecurepassword"
+    ))
+    assert (response.status_code == 500)
+
+def test_internal_server_error_handler(module_client, module_db, fake_items_error):
+    client = module_client
+
+    response = client.get('api/v1/resources')
+    assert (response.status_code == 500)
+    assert (response.json['errors'][0].get("code") == "server-error")
```

Comment:
Instead of testing the json returned in many places we should be testing that the error handler is called and then separately testing the functionality of the error handler.

Reply:
I did a quick google search and it looks like there will be a certain amount of research involved for me to learn how to assert that a function was called. Can we turn that into a separate issue and merge this as-is?

Reply:
Sounds good to me. The essential mechanics is you create a mock that you provide the ability tell if it's called. It's really helpful for decoupling your tests from the behavior you're testing.